client/state/login/actions/reboot-after-login.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import type { Dispatch } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import userFactory from 'lib/user';
+import { recordTracksEventWithClientId as recordTracksEvent } from 'state/analytics/actions';
+import { getRedirectToSanitized, isTwoFactorEnabled } from 'state/login/selectors';
+
+const user = userFactory();
+
+export const rebootAfterLogin = ( tracksEventArgs: object ) => async (
+	dispatch: Dispatch,
+	getState: () => object
+) => {
+	const redirectToSanitized = getRedirectToSanitized( getState() );
+	const twoFactorEnabled = isTwoFactorEnabled( getState() );
+
+	dispatch(
+		recordTracksEvent( 'calypso_login_success', {
+			two_factor_enabled: twoFactorEnabled,
+			...tracksEventArgs,
+		} )
+	);
+
+	// Redirects to / if no redirect url is available
+	const url = redirectToSanitized || '/';
```

Comment:
Warning: premature optimization! ;)

Could this line be deleted and we incorporate into `const url = getRedirectToSanitized( getState() ) || ''` ?

Reply:
Yeah it can actually. I basically copied the exact code layout over from the component method, but probably would have written it differently if I'd done it from scratch.